### PR TITLE
added docs, now using errorhelper

### PR DESCRIPTION
**Pull request by hugomatic**

### CODE REVIEW

File: gz-grant.html [hunk 1]
```diff
@@ -233,14 +232,59 @@
...
+                if (err.detail.error.message) {
+                  msg += ": " + err.detail.error.message
+                }
+                else {
+                  msg += ": no response"
+                }
+              }
+              else {
+                try {
+                  console.log(ajax.lastError.response)
```
Comment (chapulina) [inline gz-grant.html]: Do we need this log here? Maybe it can be `console.error`?

File: gz-grant.html [hunk 2]
```diff
@@ -233,14 +232,59 @@
...
+                  const json = typeof (x) === "object"? x: JSON.parse(x)
+                  if(json.error)
+                    msg += ": " + json.error
+                  if(json.msg)
+                    msg += ": " + json.error
+                  if(json.message)
+                    msg += ": " + json.message
+                }
+                catch(err) {
+                  msg += ajax.lastError.response
```
Comment (chapulina) [inline gz-grant.html]: Add a colon and space? Maybe we can get `lastError.response` before the try/catch?

File: gz-grant.html (hunk 1 above)
Comment (chapulina) [inline gz-grant.html]: I just saw the console.error in the end, I think this is not necessary here.

Comment (hugomatic) [reply gz-grant.html]: I removed it, it was only there for debugging.

File: gz-grant.html (hunk 2 above)
Comment (hugomatic) [reply gz-grant.html]: I added the colon and comments to try to clarify what's going on. The lastError.response is valid before the try catch, but it is used in the catch if we weren't able to get an error message.

**Comment by hugomatic** [comment]
The _errorHelper is function is copied from the gz-machinetypes. changes should be copied over when this PR is merged.

File: gz-grant.html [hunk 3]
```diff
@@ -233,14 +232,61 @@
...
+              else {
+                // if there was a response, look for an object
+                // with a message / error / msg attribute
+                try {
+                  const x = ajax.lastError.response
+                  const json = typeof (x) === "object"? x: JSON.parse(x)
+                  if(json.error)
+                    msg += ": " + json.error
+                  if(json.msg)
+                    msg += ": " + json.error
```
Comment (chapulina) [inline gz-grant.html]: json.msg

**Comment by chapulina** [comment]
+1

**Pull request merged**